core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -39,10 +64,20 @@
 
     private Integer shortCircuitCutoff;
 
-    private QueryInnerHitBuilder innerHit = null;
+    private QueryInnerHits queryInnerHits;
 
     static final HasChildQueryBuilder PROTOTYPE = new HasChildQueryBuilder(null, null);
 
+    public HasChildQueryBuilder(String type, QueryBuilder queryBuilder, Integer maxChildren, Integer minChildren, Integer shortCircuitCutoff, ScoreType scoreType, QueryInnerHits queryInnerHits) {
+        this.childType = type;
+        this.queryBuilder = queryBuilder;
+        this.scoreType = scoreType;
+        this.maxChildren = maxChildren;
+        this.minChildren = minChildren;
+        this.shortCircuitCutoff = shortCircuitCutoff;
+        this.queryInnerHits = queryInnerHits;
+    }
+
     public HasChildQueryBuilder(String type, QueryBuilder queryBuilder) {
```

Comment:
sure things change now that we know for sure the target branch, that said making everything final would be better to do once we merged back to master to prevent merge conflicts here. Same with renaming XYZQueryBuilder to XYZQuery, and moving to proper getters and setters.